ruma-serde/src/single_element_seq.rs

```diff
@@ -0,0 +1,25 @@
+//! De-/serialization functions to and from single element sequences.
+
+use serde::{
+    de::{Deserialize, DeserializeOwned, Deserializer},
+    ser::{Serialize, Serializer},
+};
+
+pub fn serialize<T, S>(element: &T, serializer: S) -> Result<S::Ok, S::Error>
+where
+    T: Serialize,
+    S: Serializer,
+{
+    [element].serialize(serializer)
+}
+
+pub fn deserialize<'de, T, D>(deserializer: D) -> Result<T, D::Error>
+where
+    T: DeserializeOwned,
+    D: Deserializer<'de>,
+{
+    <[_; 1]>::deserialize(deserializer).map(|array| {
+        let [first] = array;
+        first
+    })
```

Comment:
Function / closure arguments are actually patterns, so this can be shortened to

```suggestion
    <[_; 1]>::deserialize(deserializer).map(|[first]| first)
```